### ISSUE TITLE
Implement missing SQR1 and SQR2 functions

### ISSUE THREAD

**Issue by CloudyPadmal** [fossasia/pslab-python#197]
In the previous psl package, there were functions to control SQR1 and SQR2 pins individually. However, they are missing in the latest version and one has to use SQR4 function driving all 4 SQR pins.

With the latest firmware upgrades, it is made possible to generate sine and square waves simultaneously and we need to implement whatever the missing functions referring to the old python package.

Refer to following as a guide:
- [SQR1](https://github.com/fossasia/pslab-python/blob/420a97eaac0f54d5185ad785e09a205340f422df/PSL/sciencelab.py#L3127)
- [SQR2](https://github.com/fossasia/pslab-python/blob/420a97eaac0f54d5185ad785e09a205340f422df/PSL/sciencelab.py#L3197)

**Comment by bessman** [comment]
We will need to add a way to check the firmware version as well, because I think WAVEGEN->SET_SQR2 is broken in the current version.

**Comment by CloudyPadmal** [comment]
That's true.. in the old SQR2 function in firmware, the right timer is never turned on making it broken. I wonder if there's a way to trigger it indirectly. 🤔

Now that I'm thinking, it would be nice if we had a getter setter function to access any arbitrary register in firmware.

**Comment by SehajModi** [comment]
Hi everyone! I am a first-year ICE (Instrumentation and Control) student, and I have been studying the current set_sqr4 implementation. I’m very interested in working on adding individual control for SQR1 and SQR2 as discussed here.

Since there were concerns about broken firmware logic for SQR2, should I start by implementing the generic register getter/setter functions first, or should I attempt the specific SQR functions directly? Also, if there is a firmware register map or protocol document available, I’d love to take a look at it to better understand the hardware side. Could you please assign this to me?

**Comment by marcnause** [comment]
@bessman Is this issue still valid?
